src/components/RichTextEditor/serializer/utils/serializeLeafToHtml.ts

```diff
@@ -12,7 +12,8 @@ import escapeHtml from 'escape-html';
 
 export const serializeLeafToHtml = (node: TText): string => {
     let string = escapeHtml(node.text);
-    string = string.replaceAll('\n', '<br />');
+    string === '' && (string = '&#xFEFF;');
```

Comment:
Question: why is this needed?

Reply:
also nit, bit I was a bit confused at first when looking at this line, maybe
```suggestion
    if (string === '') {
        string = '&#xFEFF;'
    }
```

Reply:
`Question: why is this needed?`

It's needed for hard breaks, so we don't get empty h1 or other tags that are not displayed. So we ensure that the tags are always rendered.